maker/src/cfd.rs

```diff
@@ -185,6 +200,7 @@ pub struct Actor<O, T, W> {
     oracle: xtra::Address<O>,
     time_to_first_position: xtra::Address<time_to_first_position::Actor>,
     connected_takers: HashSet<Identity>,
+    taker_versions: HashMap<Identity, Option<TakerDaemonVersion>>,
```

Comment:
🙃 This was the simplest way of keeping this information where it is needed, but I am not sure I agree that it should be here. I think it would be better if we would keep the `taker_version` in the connection actor and dispatch different messages there. 
This would be mean more code, but I think it would be cleaner because we encapsulate the decision what version it is in one place. 

I am not totally against the proposed solution, but we scatter decision logic based on network versions over multiple actors which might get harder to maintain and clean up.